libs/store/src/lib/store/utils/dispatch-data-to-store/dispatch-data-to-store.util.ts

```diff
@@ -1,37 +1,159 @@
+	store: Store
+): Observable<null>;
+// ENTITY STORE
+export function dispatchDataToStore<DataType = any>(
+	dispatchAction: EntityStoreActions<DataType>,
+	selectAction: EntityStoreSelectors<DataType>,
+	data: Observable<DataType[]>,
+	store: Store,
+	dispatchType: 'set' | 'add' | 'update'
+): Observable<DataType[]>;
+export function dispatchDataToStore<DataType = any>(
+	dispatchAction: EntityStoreActions<DataType>,
+	selectAction: EntityStoreSelectors<DataType>,
+	data: Observable<DataType[]>,
+	store: Store
+): Observable<DataType[]>;
+// NO SELECTOR
+export function dispatchDataToStore<DataType = any>(
+	dispatchAction: EntityStoreActions<DataType>,
+	selectAction: null | undefined,
+	data: Observable<DataType[]>,
+	store: Store,
+	dispatchType: 'set' | 'add' | 'update'
+): Observable<null>;
+export function dispatchDataToStore<DataType = any>(
+	dispatchAction: EntityStoreActions<DataType>,
+	selectAction: null | undefined,
+	data: Observable<DataType[]>,
+	store: Store
+): Observable<null>;
+
+// -------------------
+// Definition
+// -------------------
 /**
  * Dispatches data to the store based on a provided action and Observable. Loading and error state will be handled by default.
  *
  * @param dispatchAction - The action we wish to use to dispatch data to the store
+ * @param selectAction - The selector we wish to use to select the data from the store. If not provided, `null` will be returned.
  * @param data - The data we wish to dispatch to the store
  * @param store - The store we wish to dispatch the data to
  * @param dispatchType - Whether we wish to set, add or update the data. By default this is `set`
- * @return {*}  {Observable<DataType>}
+ * @returns {Observable<(DataType | DataType[] | null)>} - Returns the data that was dispatched to the store. The generic type is the same as the `data`.
  */
-export const dispatchDataToStore = <DataType>(
-	dispatchAction: any,
-	data: Observable<DataType>,
+export function dispatchDataToStore<DataType = any>(
+	dispatchAction: BaseStoreActions<DataType> | EntityStoreActions<DataType>,
+	selectAction: BaseStoreSelectors<DataType> | EntityStoreSelectors<DataType> | undefined,
+	data: Observable<DataType | DataType[]>,
 	store: Store,
 	dispatchType: 'set' | 'add' | 'update' = 'set'
-): Observable<DataType> => {
+): Observable<DataType | DataType[]> | null {
 	// Iben: Set the loading state to true and the error state to false to start a new set
 	store.dispatch(dispatchAction.loading({ payload: true }));
 	store.dispatch(dispatchAction.error({ payload: false }));
 
 	// Iben: Set, add or update the data according to the provided dispatch type
 	return data.pipe(
 		tap((payload) => {
-			if (dispatchType === 'set') {
-				store.dispatch(dispatchAction.set({ payload }));
-			} else if (dispatchType === 'add') {
-				store.dispatch(dispatchAction.add({ payload }));
-			} else {
-				store.dispatch(dispatchAction.update({ payload }));
+			switch (dispatchType) {
```

Comment:
Given that a switch case or an if else structure in this case makes no difference, I'd keep it the way it is if it isn't necessary to change it.

Changing these kinds of things in a PR without real cause makes it harder to review PRs, because you have to first try to figure out why it was done in relation to the changes, and then realize it wasn't really needed to do so.